SimPEG/Utils/io_utils.py

```diff
@@ -146,35 +147,72 @@ def surface2inds(vrtx, trgl, mesh, boundaries=True, internal=True):
     return insideGrid
 
 
-def remoteDownload(url, remoteFiles, basePath=None):
+def remoteDownload(
+    url, remoteFiles, path='.', directory='SimPEGtemp', rm_previous=False
+):
     """
     Function to download all files stored in a cloud directory
-    var:    url ("http:\\...")
-    list:   List of file names to download
+
+    :param str url: url of the storage bucket ("http://...")
+    :param list remoteFiles: List of file names to download from the storate bucket
+    :param str path: path to where the directory is created and files downloaded (default is the current directory)
+    :param str directory: name of the directory to be created and have content downloaded to
+    :param bool rm_previous: remove file and contents if a directory with the specified name already exists
     """
 
     # Download from cloud
     import urllib
     import shutil
     import os
     import sys
+
+    def rename_path(downloadpath):
```

Comment:
if we start to make heavy use of this function, I can see a number of scenarios where we do not necessarily want to over-write the contents of `SimPEGtemp`, so here it will add `SimPEGtemp(1)` (or up the integer value until we hit a unique name, similar to what you would see with a web browser)